src/backend/distributed/worker/worker_partition_protocol.c

```diff
@@ -336,6 +337,16 @@ GetFunctionInfo(Oid typeId, Oid accessMethodId, int16 procedureId)
 {
 	FmgrInfo *functionInfo = (FmgrInfo *) palloc0(sizeof(FmgrInfo));
 
+	/* if the type is being dropped, early exit with a null return value */
+	HeapTuple tup = SearchSysCache1(TYPEOID, ObjectIdGetDatum(typeId));
+	if (!HeapTupleIsValid(tup))
```

Comment:
We should allow null heap tuples for a type only when we reach here via a drop trigger. There is no easy way to check if we are now doing a drop on the type.